packages/core/src/dialects/cockroachdb/connection-manager.ts

```diff
@@ -0,0 +1,254 @@
+import pick from 'lodash/pick';
+import type { Client, ClientConfig } from 'pg';
+import type { TypeFormat, TypeId } from 'pg-types';
+import {
+  ConnectionError,
+  ConnectionRefusedError,
+  ConnectionTimedOutError,
+  HostNotFoundError,
+  HostNotReachableError,
+  InvalidConnectionError,
+} from '../../errors';
+import type { ConnectionOptions } from '../../sequelize.js';
+import { Sequelize } from '../../sequelize.js';
+import { logger } from '../../utils/logger';
+import type { Connection } from '../abstract/connection-manager';
+import { AbstractConnectionManager } from '../abstract/connection-manager';
+import type { CockroachDbDialect } from './index';
+
+export interface CockroachdbConnection extends Connection, Client {
+  options?: string;
+  // custom property we attach to the client
+  // TODO: replace with Symbols.
+  _invalid?: boolean;
+  standard_conforming_strings?: boolean;
+
+  // Private property of pg-client
+  // TODO: ask pg to expose a stable, readonly, property we can use
+  _ending?: boolean;
+}
+
+// TODO: once the code has been split into packages, we won't need to lazy load pg anymore
+// eslint-disable-next-line @typescript-eslint/consistent-type-imports
+type Lib = typeof import('pg');
+// eslint-disable-next-line @typescript-eslint/consistent-type-imports
+type ArrayParserLib = typeof import('postgres-array');
+type TypeParser = (source: string) => unknown;
+
+interface TypeOids {
+  oid: number;
+  typeName: string;
+  type: 'base' | 'array';
+  /** oid of the base type. Available on array, range & range-array */
+  baseOid?: number;
+  /** oid of the range. Available on range-array */
+  rangeOid?: number;
+}
+
+const debug = logger.debugContext('connection:crdb');
+
+export class CockroachdbConnectionManager extends AbstractConnectionManager<CockroachdbConnection> {
+  private readonly lib: Lib;
+  readonly #arrayParserLib: ArrayParserLib;
+
+  #oidMap = new Map<number, TypeOids>();
+  #oidParserCache = new Map<number, TypeParser>();
+
+  constructor(dialect: CockroachDbDialect, sequelize: Sequelize) {
+    super(dialect, sequelize);
+    this.lib = this._loadDialectModule('pg') as Lib;
+    this.#arrayParserLib = this._loadDialectModule('postgres-array') as ArrayParserLib;
+  }
+
+  async connect(config: ConnectionOptions): Promise<CockroachdbConnection> {
+    const port = Number(config.port ?? this.dialect.getDefaultPort());
+
+    // @ts-expect-error -- "dialectOptions.options" must be a string in PG, but a Record in MSSQL. We'll fix the typings when we split the dialects into their own modules.
+    const connectionConfig: ClientConfig = {
+      ...(config.dialectOptions && pick(config.dialectOptions, [
+        // Value for application name variable see here https://www.cockroachlabs.com/docs/stable/connection-parameters.html#additional-connection-parameters
```

Comment:
These URLs don't seem relevant. They're pg options so they should link to the documentation of pg.